src/data/real/sqrt.lean

```diff
@@ -97,15 +97,98 @@ end nnreal
 namespace real
 
 /-- An auxiliary sequence of rational numbers that converges to `real.sqrt (mk f)`.
+This is Heron's method for computing square roots.
 Currently this sequence is not used in `mathlib`.  -/
 def sqrt_aux (f : cau_seq ℚ abs) : ℕ → ℚ
-| 0       := rat.mk_nat (f 0).num.to_nat.sqrt (f 0).denom.sqrt
-| (n + 1) := let s := sqrt_aux n in max 0 $ (s + f (n+1) / s) / 2
+| 0       := 1
+| (n + 1) := let s := sqrt_aux n in (s + (max 0 (f (n+1))) / s) / 2
 
-theorem sqrt_aux_nonneg (f : cau_seq ℚ abs) : ∀ i : ℕ, 0 ≤ sqrt_aux f i
-| 0       := by rw [sqrt_aux, rat.mk_nat_eq, rat.mk_eq_div];
-  apply div_nonneg; exact int.cast_nonneg.2 (int.of_nat_nonneg _)
-| (n + 1) := le_max_left _ _
+theorem sqrt_aux_nonneg (f : cau_seq ℚ abs) (i : ℕ): 0 ≤ sqrt_aux f i :=
+begin
+  induction i with i hyp,
+  { unfold sqrt_aux, exact zero_le_one},
+  { unfold sqrt_aux,
+    simp, cancel_denoms,
```

Comment:
You shouldn't leave a plain `simp` if it doesn't prove the goal - use `simp?` or `squeeze_simp` to replace it with `simp only [...]` with just the lemmas you use